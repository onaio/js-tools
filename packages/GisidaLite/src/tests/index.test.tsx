import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import ReactMapboxGl from 'react-mapbox-gl';
import {
  arePropsEqual,
  gisidaLiteDefaultProps,
  GisidaLiteProps,
  MemoizedGisidaLite,
  ReactMapboxGlProps
} from '..';

jest.mock('react-mapbox-gl');

describe('component/GisidaLite', () => {
  it('renders without crashing', () => {
    shallow(<MemoizedGisidaLite {...gisidaLiteDefaultProps} />);
  });
  it('renders props correctly', () => {
    const wrapper = mount(<MemoizedGisidaLite {...gisidaLiteDefaultProps} />);
    expect(wrapper.props()).toMatchSnapshot('gisida lite props');
  });
  it('renders component correctly', () => {
    const wrapper = mount(<MemoizedGisidaLite {...gisidaLiteDefaultProps} />);
    expect(toJson(wrapper)).toMatchSnapshot('gisida lite render');
  });
  it('reacts to props changes accordingly in relation to ReactMapboxGl constructor', () => {
    const updatedReactMapboxglProps = {
      accessToken: '',
      attributionControl: true,
      customAttribution: '',
      injectCSS: true
    };
    const mockReactMapboxGlMock = jest.fn();
    (ReactMapboxGl as jest.Mock).mockImplementationOnce(() => mockReactMapboxGlMock);

    const wrapper = mount(<MemoizedGisidaLite {...gisidaLiteDefaultProps} />);
    expect(mockReactMapboxGlMock).toHaveBeenCalledTimes(1);
    expect(wrapper.props().reactMapboxGlMapFactoryUtilConfigs).toEqual(ReactMapboxGlProps);
    // should not re-instanciate reactmapboxgl on other props change
    wrapper.setProps({
      ...gisidaLiteDefaultProps,
      layers: []
    });
    expect(mockReactMapboxGlMock).toHaveBeenCalledTimes(1);
    expect(wrapper.props().reactMapboxGlMapFactoryUtilConfigs).toEqual(ReactMapboxGlProps);
    // should re-instanciate map on reactmapboxgl factory options change
    wrapper.setProps({
      ...gisidaLiteDefaultProps,
      reactMapboxGlMapFactoryUtilConfigs: updatedReactMapboxglProps
    });
    expect(wrapper.props().reactMapboxGlMapFactoryUtilConfigs).toEqual(updatedReactMapboxglProps);
  });
  it('returns false if layers length has changed', () => {
    const prevProps = {
      ...gisidaLiteDefaultProps,
      layers: []
    };
    const nextProps = {
      ...gisidaLiteDefaultProps
    };
    expect(arePropsEqual(prevProps, nextProps)).toEqual(false);
  });

  it('returns true if length has not changed', () => {
    const prevProps: GisidaLiteProps = {
      ...gisidaLiteDefaultProps,
      layers: []
    };
    const nextProps: GisidaLiteProps = {
      ...gisidaLiteDefaultProps,
      layers: []
    };
    expect(arePropsEqual(prevProps, nextProps)).toEqual(true);
  });
});
