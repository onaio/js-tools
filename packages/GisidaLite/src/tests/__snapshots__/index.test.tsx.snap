// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/GisidaLite renders component correctly: gisida lite render 1`] = `
<GisidaLite
  layers={
    Array [
      <MappedComponent
        data={
          Object {
            "geometry": Object {
              "coordinates": Array [
                0,
                0,
              ],
              "type": "Point",
            },
            "type": "Feature",
          }
        }
        symbolLayout={
          Object {
            "icon-image": "triangle-15",
          }
        }
      />,
    ]
  }
  mapComponents={
    Array [
      <MappedComponent
        position="top-right"
      />,
      <MappedComponent
        position="bottom-left"
      />,
    ]
  }
  mapConfigs={
    Object {
      "containerStyle": Object {
        "height": "100vh",
        "width": "100vw",
      },
      "style": "mapbox://styles/mapbox/streets-v9",
    }
  }
  reactMapboxGlMapFactoryUtilConfigs={
    Object {
      "accessToken": "pk.eyJ1Ijoib25hIiwiYSI6IlVYbkdyclkifQ.0Bz-QOOXZZK01dq4MuMImQ",
      "attributionControl": true,
      "customAttribution": "&copy; ONA Engineering",
      "injectCSS": true,
    }
  }
/>
`;

exports[`component/GisidaLite renders props correctly: gisida lite props 1`] = `
Object {
  "layers": Array [
    <MappedComponent
      data={
        Object {
          "geometry": Object {
            "coordinates": Array [
              0,
              0,
            ],
            "type": "Point",
          },
          "type": "Feature",
        }
      }
      symbolLayout={
        Object {
          "icon-image": "triangle-15",
        }
      }
    />,
  ],
  "mapComponents": Array [
    <MappedComponent
      position="top-right"
    />,
    <MappedComponent
      position="bottom-left"
    />,
  ],
  "mapConfigs": Object {
    "containerStyle": Object {
      "height": "100vh",
      "width": "100vw",
    },
    "style": "mapbox://styles/mapbox/streets-v9",
  },
  "reactMapboxGlMapFactoryUtilConfigs": Object {
    "accessToken": "pk.eyJ1Ijoib25hIiwiYSI6IlVYbkdyclkifQ.0Bz-QOOXZZK01dq4MuMImQ",
    "attributionControl": true,
    "customAttribution": "&copy; ONA Engineering",
    "injectCSS": true,
  },
}
`;
